web/src/components/users/RootAuthMethodsPage.tsx

```diff
@@ -0,0 +1,168 @@
+
+import React, { useRef, useState } from "react";
+import {
+  Button,
+  FileUpload,
+  Flex,
+  Form,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+} from "@patternfly/react-core";
+import { useLocation, useNavigate } from "react-router-dom";
+import { Center } from "~/components/layout";
+import { Page, PasswordInput } from "~/components/core";
+import { useRootUserMutation } from "~/queries/users";
+import { RootUserChanges } from "~/types/users";
+import { ROOT as PATHS } from "~/routes/paths";
+import { isEmpty } from "~/utils";
+import { _ } from "~/i18n";
+import textStyles from "@patternfly/react-styles/css/utilities/Text/text";
+import sizingStyles from "@patternfly/react-styles/css/utilities/Sizing/sizing";
+
+/**
+ * A page component for setting at least one root authentication method
+ *
+ * NOTE: This page will be automatically displayed only when no root authentication
+ * method is set. It is not within the scope of this component to fill data if
+ * users manually enter the route path.
+ */
+function RootAuthMethodsPage() {
+  const passwordRef = useRef();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const setRootUser = useRootUserMutation();
+  const [password, setPassword] = useState("");
+  const [sshKey, setSSHKey] = useState("");
+  const [isUploading, setIsUploading] = useState(false);
+
+  const startUploading = () => setIsUploading(true);
+  const stopUploading = () => setIsUploading(false);
+  const clearKey = () => setSSHKey("");
+
+  const isFormValid = !isEmpty(password) || !isEmpty(sshKey);
+  const uploadFile = () => document.getElementById("sshKey-browse-button").click();
+
+  const accept = async (e: React.SyntheticEvent) => {
+    e.preventDefault();
+
+    const data: Partial<RootUserChanges> = {};
+
+    if (!isEmpty(password)) {
+      data.password = password;
+      data.encryptedPassword = false;
+    }
+
+    if (!isEmpty(sshKey)) {
+      data.sshkey = sshKey;
+    }
+
+    if (isEmpty(data)) return;
+
+    await setRootUser.mutateAsync(data);
+
+    navigate(location.state?.from || PATHS.root, { replace: true });
+  };
+
+  // TRANSLATORS: %s will be replaced by a link with the text "upload".
+  const [sshKeyStartHelperText, sshKeyEndHelperText] = _(
+    "Write, paste, drop, or %s a SSH public key file in the above textarea.",
+  ).split("%s");
+  // TRANSLATORS: this "upload" is a commanding verb, in the %s place of
+  // "Write, paste, drop, or %s a SSH public key file in the above textarea."
+  const uploadLinkText = _("upload");
+
+  return (
+    <Page>
+      <Page.Content>
+        <Center>
+          <Page.Section
+            headerLevel="h2"
+            title={_("Setup root user authentication")}
+            description={
+              <p className={textStyles.fontSizeXl}>
+                {_(
+                  "You must define at least one authentication method for the root user. You can still edit them anytime before the installation.",
```

Comment:
> will the new root password screen be localizable?

@bittin yes of course, as you can see here.
If by any chance it does not work then we may have broken things in the recent refactoring, #1777.

Reply:
@mvidner ah alright thanks, have not seen anything pop up at: https://l10n.opensuse.org/projects/agama/-/sv/ yet